carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -216,14 +232,14 @@ protected boolean ensureCredentials() {
 
             if (url != null) {
                 try {
-                    amazonCredentials = new PropertiesCredentials(url.openStream());
+					amazonCredentials = getRemoteFileCredentials(url);
                     return true;
                 } catch (Exception e) {
                     // fail out
                 }
             } else {
                 try {
-                    amazonCredentials = new PropertiesCredentials(new File(getCredentials()));
+                    amazonCredentials = new PropertiesFileCredentialsProvider(getCredentials());
```

Comment:
Updated this to use PropertiesFileCredentialsProvider if the file exists on the system, if it's remote I load the properties and created an in-line provider implementation to just return a BasicAwsCredentials object (removing the internal reference to StaticPropertiesProvider) - let me know what you think